src/Pages/PeoplePage.tsx

```diff
@@ -0,0 +1,74 @@
+import { PeopleFilters } from '../components/PeopleFilters';
+import { useState, useEffect } from 'react';
+import { PeopleTable } from '../components/PeopleTable/PeopleTable';
+import { Loader } from '../components/Loader';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { filterPeople } from '../utils/filterPeople';
+import { sortPeople } from '../utils/SortPeople';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const { slug } = useParams();
+
+  useEffect(() => {
+    getPeople()
+      .then(data => {
+        setPeople(data);
+        setLoading(false);
+      })
+      .catch(() => {
+        setError(true);
+        setLoading(false);
+      });
+  }, []);
+
+  const [searchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+
+  const filteredPeople = filterPeople(people, { sex, query, centuries });
+  const sortedPeople = sortPeople(filteredPeople, { sort, order });
```

Comment:
you can wrap both these variables into useMemo to avoid filtering when sort option changes and vice versa:

```suggestion
  const filteredPeople = useMemo(() => filterPeople(people, { sex, query, centuries }), [// write dependencies here]);
  const sortedPeople = sortPeople(filteredPeople, { sort, order }); // same for this variable
```